samples/dnn/segmentation.cpp

```diff
@@ -146,29 +146,21 @@ int main(int argc, char **argv)
         blobFromImage(frame, blob, scale, Size(inpWidth, inpHeight), mean, swapRB, false);
         //! [Set input blob]
         net.setInput(blob);
-        //! [Make forward pass]
-        Mat score = net.forward();
+        //! [Set input blob]
+
         if (modelName == "u2netp")
         {
-            Mat mask, thresholded_mask, foreground_overlay, background_overlay, foreground_segmented;
-            mask = cv::Mat(score.size[2], score.size[3], CV_32F, score.ptr<float>(0, 0));
-            mask.convertTo(mask, CV_8U, 255);
-            threshold(mask, thresholded_mask, 0, 255, THRESH_BINARY + THRESH_OTSU);
-            resize(thresholded_mask, thresholded_mask, Size(frame.cols, frame.rows), 0, 0, INTER_AREA);
-            // Create overlays for foreground and background
-            foreground_overlay = Mat::zeros(frame.size(), frame.type());
-            background_overlay = Mat::zeros(frame.size(), frame.type());
-            // Set foreground (object) to red and background to blue
-            foreground_overlay.setTo(Scalar(0, 0, 255), thresholded_mask);
-            Mat inverted_mask;
-            bitwise_not(thresholded_mask, inverted_mask);
-            background_overlay.setTo(Scalar(255, 0, 0), inverted_mask);
-            // Blend the overlays with the original frame
-            addWeighted(frame, 1, foreground_overlay, 0.5, 0, foreground_segmented);
-            addWeighted(foreground_segmented, 1, background_overlay, 0.5, 0, frame);
```

Comment:
Why do you remove the solution with overlays? It looks better, IMHO.

Reply:
@asmorkalov The models.yml preprocessing pararmeters have been fixed. For overlay I have added the solution back but without thresholding. The model does very well in fine areas, the effect is lost with thresholding. This is how it looks now

![hockey](https://github.com/opencv/opencv/assets/25356067/65e51836-b3c9-4ad4-a102-36ef6744a959)
![horse](https://github.com/opencv/opencv/assets/25356067/25193006-eb22-4faa-84a0-eee82018e703)